### PR TITLE
Fixed subtraction and reference multiSelect lists rendering improperly when the list is empty

### PR DESCRIPTION
**Pull request by ReeceHoffmann**
Changes:

1. After a few revisions ended up fixing VIR-948 by having the SubtractionSelector component directly return a NoneFoundBox if there are no subtractions to be listed. 
2. Added similar code to ReferenceSelector to create a properly formed error message
3. NoneFoundBox was modified to accept and directly render props.children to permit inclusion of links to the correct pages. Assumes that the child will be an inline element otherwise formatting will be off
4. Changed the content checking ternary operator in MultiSelector so that it checks the truthiness of children.length rather than children

### CODE REVIEW

File: client/src/js/subtraction/components/Selector.js [hunk 1]
```diff
@@ -2,14 +2,24 @@ import { map } from "lodash-es";
...
     const subtractionComponents = map(subtractions, ({ name, id }) => (
         <MultiSelectorItem key={id} name={name} value={id} id={id}>
             <span>{name}</span>
         </MultiSelectorItem>
     ));
-
+    if (!subtractions.length) {
+        return (
+            <NoneFoundBox noun={"subtractions"}>
+                <BrowserRouter>
```
Comment (igboyes) [inline client/src/js/subtraction/components/Selector.js:17]: Why did you need `BrowserRouter` here?

File: client/src/js/subtraction/components/Selector.js [hunk 2]
```diff
@@ -2,14 +2,24 @@ import { map } from "lodash-es";
...
 export const SampleSubtractionSelector = ({ name, noun, selected, subtractions, onChange }) => {
     const subtractionComponents = map(subtractions, ({ name, id }) => (
         <MultiSelectorItem key={id} name={name} value={id} id={id}>
             <span>{name}</span>
         </MultiSelectorItem>
     ));
-
+    if (!subtractions.length) {
+        return (
+            <NoneFoundBox noun={"subtractions"}>
```
Comment (igboyes) [inline client/src/js/subtraction/components/Selector.js:16]:
```suggestion
            <NoneFoundBox noun="subtractions">
```

**Review by igboyes — CHANGES_REQUESTED** [review]

File: client/src/js/subtraction/components/Selector.js (hunk 1 above)
Comment (ReeceHoffmann) [reply client/src/js/subtraction/components/Selector.js]: Turns out I don't need it there. Jest was complaining that the Link wasn't wrapped while running test because it lacked the normal context of the router from the parent components. Fixed in latest version by wrapping the whole component in the test rather than adding a new router to the production code.

**Review by igboyes — APPROVED** [review]
